semantic_model_generator/snowflake_utils/snowflake_connector.py

```diff
@@ -74,6 +78,49 @@
 _QUERY_TAG = "SEMANTIC_MODEL_GENERATOR"
 
 
+def _get_table_comment(
+    conn: SnowflakeConnection, table_name: str, columns_df: pd.DataFrame
+) -> str:
+    if columns_df[_TABLE_COMMENT_COL].iloc[0]:
+        return columns_df[_TABLE_COMMENT_COL].iloc[0]  # type: ignore[no-any-return]
+    else:
+        # auto-generate table comment if it is not provided.
+        try:
+            tbl_ddl = (
+                conn.cursor()  # type: ignore[union-attr]
+                .execute(f"select get_ddl('table', '{table_name}');")
+                .fetchall()[0][0]
+            )
+            comment_prompt = f"Here is a table with below DDL: {tbl_ddl} \nPlease provide a business description for the table. Only return the description without any other text."
+            complete_sql = f"select SNOWFLAKE.CORTEX.COMPLETE('{_autogen_model}', '{comment_prompt}')"
+            cmt = conn.cursor().execute(complete_sql).fetchall()[0][0]  # type: ignore[union-attr]
+            return str(cmt + AUTOGEN_TOKEN)
+        except Exception as e:
+            logger.warning(f"Unable to auto generate table comment: {e}")
+            return ""
```

Comment:
should we make this a flag that the user controls?

Reply:
Thought about it, but also don't see much cases user would not want to generate